### PR TITLE
FadeIn component

### PR DESCRIPTION
**Pull request by mdodgelooker**
### :sparkles: Changes

- Adds a base component `Animate` and a `FadeIn` component that wraps it with opacity keyframes
- Currently the only use case in the core product is for `FadeIn`
- An "out" transition (e.g. `FadeOut`) could be done with `useAnimationState` but I think that can be left for a future PR / when there's a use case

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### :camera: Screenshots
2818b5dd-f0ae-4b5a-94d0-0f9d2854f454

### CODE REVIEW

File: packages/components/src/Animate/Animate.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  animation-duration: ${({ duration = 'quick', theme }) =>
+    `${theme.transitions[duration]}ms`};
+`
+
+export const Animate = styled((props: AnimateProps) => (
+  <div {...omit(props, ['delay', 'duration'])} />
+))`
+  ${simpleLayoutCSS}
+  ${animateCSS}
+`
```
Comment (mdodgelooker) [inline packages/components/src/Animate/Animate.tsx:62]: This component isn't particularly useful as is... should it even be exported?

File: packages/components/src/Animate/Animate.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+export const FadeIn = styled(Animate)`
+  animation-fill-mode: forwards;
+  animation-name: ${fadeIn};
+  opacity: 0;
+`
```
Comment (mdodgelooker) [inline packages/components/src/Animate/Animate.tsx:68]: I couldn't decide on a testing strategy. An image snapshot won't work, but I hesitate to add more DOM snapshots, though that would capture the relevant CSS, beyond which there's nothing much to test. Thoughts?

Comment (lukelooker) [reply packages/components/src/Animate/Animate.tsx:68]: I think a snapshot would at least point at the CSS being stable / the component rendering (not throwing an error) so I'd be good with that for now :)

File: packages/components/src/Animate/Animate.tsx (hunk 1 above)
Comment (lukelooker) [reply packages/components/src/Animate/Animate.tsx:62]: I'm not sure... I suppose it gives easy access to the theme.durations if someone wanted to build their own FadeX or other animation component.

I'm ambivalent honestly.

File: packages/components/src/Animate/Animate.tsx (hunk 2 above)
Comment (lukelooker) [reply packages/components/src/Animate/Animate.tsx:68]: It just occurred to me that you might be able to use the Jest timers to measure opacity at the beginning of the duration, then check that it's resolved at the end. That would "prove" that the animation ends in a good state I think?

**Review by lukelooker — CHANGES_REQUESTED** [review]
I think figuring out some kind of test (even if just snapshot is viable) would be the only addition I'd like to see.

File: packages/components/src/Animate/Animate.tsx (hunk 2 above)
Comment (mdodgelooker) [reply packages/components/src/Animate/Animate.tsx:68]: I added a couple snapshots (default props & non-default), but sadly the jest timers don't seem to have any affect on the css values.

**Review by lukelooker — APPROVED** [review]

**Pull request merged**